### PR TITLE
remove reference viscosity from the material model interface

### PR DESCRIPTION
**Pull request by jdannberg**
This PR removes the reference viscosity from the material model interface since we do not require it any more for the pressure scaling (or anywhere else in ASPECT). 

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: doc/modules/changes/20220520_jdannberg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Changed: The reference viscosity function is no longer part
+of the material model interface (because we do not require 
+it anymore for the pressure scaling or anywhere else). The
+One material model affected by this is the depth-dependent
```
Comment (bangerth) [inline doc/modules/changes/20220520_jdannberg:4]:
```suggestion
one material model affected by this is the depth-dependent
```

File: include/aspect/material_model/composition_reaction.h [hunk 2]
```diff
@@ -74,7 +74,7 @@ namespace aspect
          * @name Reference quantities
          * @{
          */
-        double reference_viscosity () const override;
+        double reference_viscosity () const;
```
Comment (bangerth) [inline include/aspect/material_model/composition_reaction.h:77]: Is this still called somewhere?

**Review by bangerth — COMMENTED** [review]
Yay. Long time to get rid of this.

In all of these material models, you leave these functions in place, just no longer `virtual`. But who calls them? Are they still really needed?

**Comment by jdannberg** [comment]
Most of these are probably no longer needed. I asked @gassmoeller and he suggested to remove the ones that are not needed in a follow-up PR so that this patch is shorter.

**Comment by jdannberg** [comment]
It looks like there already is an issue (#4339) that says to do this in two steps. So this PR addresses part one of the issue.

**Comment by jdannberg** [comment]
I have also addressed your one comment.

**Comment by bangerth** [comment]
Open an issue so we don't forget, fix the typo, and let's get this merged.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**